### ISSUE TITLE
Incomplete or incorrect expansion of non-ASCII characters in Thunderbird

### ISSUE THREAD

**Issue by yetiinabox** [espanso/espanso#1343]
**Describe the bug**
Espanso fails to expand various character sequences in Thunderbird on Kubuntu 22.04.1. This is a new behaviour that was not present in a 20.04 installation, with an earlier version of Thunderbird, and it does not occur outside of Thunderbird. It occurs regardless of the options used in an application-specific configuration (e.g., backend is auto, clipboard, or inject).

**To Reproduce**
1. Create expansion for email signature that includes unusual characters, such as — or ṛ
2. Trigger expansion in Thunderbird
3. Expansion fails in various ways - incomplete, or with odd sequences such as /r instead of correct expansion
e.g., 
  - trigger: ;testy
    replace: |
      kṣīrsāgare mastyāḥ krīḍanti

yields

k/s}irs}agare masty}a/h kr}i/danti

and 

  - trigger: ;bb
    replace: |
      Be well,

      —WBTD.

yields

Be well,

(with line break but nothing more).

**Expected behavior**
Expansion should occur normally.

**Screenshots**
If applicable, add screenshots or screen recordings to help explain your problem.

**Logs**
If possible, run `espanso log` in a terminal after the bug has occurred, then post the output here so that we can better diagnose the problem

Sometimes the logs show an error as follows;
12:26:52 [worker(9346)] [INFO] using X11ProxyInjector
12:26:53 [worker(9346)] [INFO] using X11Clipboard
12:41:29 [worker(9346)] [ERROR] text injector (event) reported an error: missing vkey mapping for char `—`

...but sometime they show no error at all (as in the mātsyāh example)

**Setup information**
- OS: Kubuntu 22.0.4.1
- Version: espanso 2.1.6-beta

**Comment by michaelaye** [comment]
I got the same problem I think, with the german accents:

1. type in o" to trigger getting an ö
2. Espanso removes o" , get's stuck a bit and then triggers a Paste shortcut.

My system:
![image](https://user-images.githubusercontent.com/69774/188400620-e43c35c1-3bcc-4c24-ad40-6ba845fd79a7.png)
* espanso 2.1.4-beta

**Comment by michaelaye** [comment]
Interestingly, I get the same behavior in using Exchange email in this browser (Firefox), while this browser did above umlauts in Github just fine.

**Comment by michaelaye** [comment]
And then I tried it in Chromium browser and things work there fine!
So it's a Mozilla problem!

**Comment by pagiraud** [comment]
I think it’s also clipboard related, since when trying to replace a string with a non-ASCII character in Thunderbird, I may end up with the content of my "quick" clipboard (previously selected text that can be pasted with a middle-click).

**Comment by sphh** [comment]
At the moment pasting into Thunderbird's (102.2.2) plain text _and_ HTML editor (even with non-ASCII characters) works for me with the following settings on Linux, X-Windows (see #455):
```yaml
filter_class: "[Tt]hunderbird"
backend: Clipboard
x11_use_xclip_backend: true
```
Please report back, if this is working for you.

**Comment by pagiraud** [comment]
I just noticed this bug only occurs within the HTML editor: if I use the plain text editor or if I try an expansion in a field (To, Object, etc.), it works. Is it the same for you?

**Comment by sphh** [comment]
For a (lengthy) investigations into html vs. plain text editors, please see #455. And you are right, it only occurs in the html editor, but not the plain text editor or any of the other input fields.

I have these lines in a separate file in `~/.config/espanso/config/` … BTW no need to restart espanso; it should pick up the changes automatically.

And most important: Which OS are you using? `x11_use_xclip_backend` obviously will only work in X11!

**Comment by pagiraud** [comment]
Thanks so much. :tada: :partying_face: By putting these three lines in the config directory and not the match directory (I created a thunderbird.yml file but it also works in default.yml) it works perfectly.
I use Thunderbird everyday and it would have been a show stopper for espanso if he couldn’t have work properly in this software.

So, to sum up:
To solve this problem (solution from @sphh ), add these three lines in `~/.config/espanso/config/default.yml`: 
```
filter_class: "[Tt]hunderbird"
backend: Clipboard
x11_use_xclip_backend: true
```

**Comment by pagiraud** [comment]
Although, there is a drawback: multi-line expansions don’t work properly. Everything appears on the same line in Thunderbird (it is ok in other apps).

**Comment by sphh** [comment]
You are absolutely right (https://github.com/espanso/espanso/issues/455#issuecomment-769297734):

> If xclip could support multiple paste formats (last time I searched it wasn't clear how to do it), we could instruct espanso to copy both the Text and HTML versions in the clipboard and let the text editor decide which one was best suited.

**Comment by tomatic** [comment]
This Thunderbird-specific configuration works perfectly with HTML matches for me, but it pastes Chinese characters in a plain-text Thunderbird email. If I disable `x11_use_xclip_backend`, it works in plain-text, but exhibits the known behaviour in a rich-text email (multi-second delay and then it pastes the previous clipboard contents).

This makes espanso unusable in Thunderbird.

**Comment by smeech** [comment]
I only got around this by having separate triggers for the HTML and plain-text compose windows.
